### ISSUE TITLE
system proxy support

### ISSUE THREAD

**Issue by helphi** [shadowsocks/go-shadowsocks2#45]
Because i am in a corporation network, i run shadowsocks with `HTTP_PROXY=192.168.1.100:80 go-shadowsocks2 -verbose -socks :1081 -c server:port -cipher aes-256-cfb -password xxxxxx`, but the system proxy `HTTP_PROXY=192.168.1.100:80` not work. Hope shadowsocks can support system proxy.

**Comment by riobard** [comment]
Do you mean in your corp network you have to use the HTTP proxy at 192.168.1.100:80 to access Internet?

**Comment by helphi** [comment]
yes, all right.
I use [shadowsocks-windows](https://github.com/shadowsocks/shadowsocks-windows) on windows, and i can set my corp proxy with `Forward Proxy...` with the GUI. So i hope can do something like this on my linux system with go-shadowsocks2.

**Comment by riobard** [comment]
Seems Windows system proxy only supports HTTP protocol, while Shadowsocks speaks SOCKS protocol. In which case this issue is a duplicate of #27.

**Comment by nadoo** [comment]
@helphi 
try: https://github.com/nadoo/glider
glider -l :1081 -f http://192.168.1.100:80,ss://AES-256-CFB:pass@server:port -v

**Comment by helphi** [comment]
I don't think this issue is duplicate of #27 , #27 is about go-ss2 act as a httpproxy, but this issue is about go-ss2 client how to use a httpproxy to connect with ss server.

I can use proxychains to let ss client connect with ss server through httpproxy, for example, on ubuntu bionic:

```bash
sudo su

apt install proxychains shadowsocks

cat << EOF > /etc/proxychains.conf
[ProxyList]
http 192.168.1.100 80
EOF

proxychains sslocal -b 127.0.0.1 -l 1080  -s 192.241.199.18 -p 12554 -k "ss8.pm-31958873"
```

then, i can set my browser use 127.0.0.1:1080 as a socks5 proxy, and sslocal connect ssserver 192.241.199.18:12554 through the httpproxy 192.168.1.100:80.

This method is not work with go-ss2, but sslocal is ok.
